course7/homeworks/waleed/ls/main.cc

```diff
@@ -1,16 +1,128 @@
 #include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
 
+#include <string>
+#include <sstream>
 #include <iostream>
 
-int main(int argc, char** argv)
+namespace bpo = boost::program_options;
+namespace bfs = boost::filesystem;
+
+unsigned int fileSize(const bfs::path& path)
 {
-	boost::filesystem::path currPath = boost::filesystem::current_path();
-	boost::filesystem::directory_iterator currDir(currPath);
+	if (!bfs::exists(path))
+		throw std::runtime_error("File doesn't exist");
+
+	if (bfs::is_regular_file(path))
```

Comment:
```is_regular_file``` will return false also for files that are not directories, like device nodes or symlinks.